backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -137,7 +138,25 @@ private void fetchCredentialsAndInitializeS3ClientWithTemporaryCredentials(Regio
     }
   }
 
-  private void initializeS3ClientWithTemporaryCredentials(Regions clientRegion) {
+  private synchronized void initializeS3ClientWithTemporaryCredentials(
+      Regions clientRegion, AmazonServiceException ex) {
+
+    if (ex == null && s3Client != null) {
+      try {
+        s3Client.doesBucketExistV2(bucketName);
+        return;
+      } catch (AmazonServiceException e) {
+        logAmazonServiceExceptionErrorCodes(e);
+        ex = e;
+      } catch (SdkClientException e) {
+        LOGGER.warn(e.getMessage());
+      }
+    }
+
+    if (ex != null && ex.getErrorCode().equals(ModelDBConstants.EXPIRED_TOKEN)) {
```

Comment:
comment for conrado's review: if we get an exception and it indicates an expired token , then we fetch the new credentials. [we override the scheduled job to fetch the token because things have gone wrong some where]

Reply:
At the start, `ex` is null, right? So how do we know `temporarySessionCredentials` has valid credentials?

Reply:
Also, what happens if the exception it not null and not expired token?